### PR TITLE
fix: reset metrics when builder re-initialized.

### PR DESCRIPTION
**Pull request by andysim3d**
[Closes/Fixes] #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: crates/builder/src/bundle_sender.rs [hunk 1]
```diff
@@ -331,6 +331,7 @@ where
                 error!("Bundle send error {error:?}");
                 self.metrics.bundle_txns_failed.increment(1);
                 state.bundle_error(error);
+                state.transaction_tracker.reset().await;
```
Comment (dancoombs) [inline crates/builder/src/bundle_sender.rs:334]: Good catch. Should we put this in the `bundle_error` function so we don't forget it? Or do you think it makes more sense here?

**Review by dancoombs — APPROVED** [review]

**Comment by dancoombs** [comment]
Do we want to backport this to v0.4?

**Review by 0xfourzerofour — APPROVED** [review]

File: crates/builder/src/bundle_sender.rs (hunk 1 above)
Comment (andysim3d) [reply crates/builder/src/bundle_sender.rs:334]: I tried but bundler_error is not an async function so I can't apply there.

**Comment by andysim3d** [comment]
> Do we want to backport this to v0.4?

will do!

**Pull request merged**